pkg/cmd/pulumi/pulumi.go

```diff
@@ -918,3 +922,57 @@ func isDevVersion(s semver.Version) bool {
 	devRegex := regexp.MustCompile(`\d*-g[0-9a-f]*$`)
 	return !s.Pre[0].IsNum && devRegex.MatchString(s.Pre[0].VersionStr)
 }
+
+// We want to expose all the flags on all the commands as configurable with
+// environment variables. To do this, we walk the Cobra command tree, and
+// declare an environment variable for each flag. Because the Cobra arguments
+// have some basic type information, we can use it to do things like accepting 1
+// and 0 as boolean values.
+func DeclareFlagsAsEnvironmentVariables(cmd *cobra.Command) {
+	convertToEnvironmentVariable := func(name string) string {
+		name = strings.ReplaceAll(name, "-", "_")
+		name = "OPTION_" + strings.ToUpper(name)
+
+		return name
+	}
+
+	exposeAsEnvironmentVariable := func(f *pflag.Flag) {
+		name := convertToEnvironmentVariable(f.Name)
+
+		var env declared.Value
+		switch f.Value.Type() {
```

Comment:
There's `cmd.PersistentFlags().StringArrayP` in some places, that might need special handling, or maybe the parsing of an array happens correctly inside cobra anyway?

Reply:
At the very least, we need a test for it. I'm honestly not sure what the right syntax for passing an array would be here. Probably just comma separated?